core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -47,6 +48,11 @@ private[spark] class CoarseGrainedExecutorBackend(
   var executor: Executor = null
   @volatile var driver: Option[RpcEndpointRef] = None
 
+  // This is a thread-local in case we ever decide to change this to a non-thread-safe RpcEndpoint
+  private[this] val ser: ThreadLocal[SerializerInstance] = new ThreadLocal[SerializerInstance] {
```

Comment:
actually now i think about it -- i think if the underlying thread pool keeps creating new threads (which it might), this thread local variable might lead to memory leak. maybe the best way to handle this is with your old one, and add a comment to the beginning of the class saying if we ever change it to support multiple threads, change here as well.